ios/RCCommonFunctionality.m

```diff
@@ -181,31 +223,63 @@ + (void)makeDeferredPurchase:(RCDeferredPromotionalPurchaseBlock)deferredPurchas
             completion(nil, [self payloadForError:error withExtraPayload:@{@"userCancelled": @(userCancelled)}]);
         } else {
             completion(@{
-                @"purchaserInfo": purchaserInfo.dictionary,
-                @"productIdentifier": transaction.payment.productIdentifier
+                    @"purchaserInfo": purchaserInfo.dictionary,
+                    @"productIdentifier": transaction.payment.productIdentifier
             }, nil);
         }
     });
 }
 
 + (void)checkTrialOrIntroductoryPriceEligibility:(nonnull NSArray<NSString *> *)productIdentifiers
-                                 completionBlock:(void (^)(NSDictionary<NSString *, NSDictionary *> *))completion
+                                 completionBlock:(RCReceiveIntroEligibilityBlock)completion
 {
     NSAssert(RCPurchases.sharedPurchases, @"You must call setup first.");
-    
+
     [RCPurchases.sharedPurchases checkTrialOrIntroductoryPriceEligibility:productIdentifiers completionBlock:^(NSDictionary<NSString *,RCIntroEligibility *> * _Nonnull dictionary) {
         NSMutableDictionary *response = [NSMutableDictionary new];
         for (NSString *productID in dictionary) {
             RCIntroEligibility *eligibility = dictionary[productID];
             response[productID] = @{
-                @"status": @(eligibility.status),
-                @"description": eligibility.description
+                    @"status": @(eligibility.status),
+                    @"description": eligibility.description
             };
         }
         completion([NSDictionary dictionaryWithDictionary:response]);
     }];
 }
 
++ (void)paymentDiscountForProductIdentifier:(NSString *)productIdentifier
+                                   discount:(nullable NSString *)discountIdentifier
+                            completionBlock:(RCHybridResponseBlock)completion
+{
+    if (@available(iOS 12.2, *)) {
+        NSAssert(RCPurchases.sharedPurchases, @"You must call setup first.");
+        [self productWithIdentifier:productIdentifier completionBlock:^(SKProduct *_Nullable aProduct) {
+            if (aProduct) {
+                SKProductDiscount *discountToUse = [self discountWithIdentifier:discountIdentifier forProduct:aProduct];
```

Comment:
what happens if this gets called from old iOS? does the method no-op because API isn't available? or does it fail in compilation?

Reply:
The check is in line 255 and the method will complete with a nil, which eventually in react native will resolve the promise with undefined.

Reply:
my bad ✋ I'd missed the other check 👍